<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="base_name"         default="audio"/>
  <arg name="data_dir"          default="/mnt/md0/gear_data"/>
  <arg name="image_extn"        default=".bag"/>
  <arg name="start_manager"     default="true"/>  
  <arg name="image_type"        default="audio"/>
  <arg name="image_prefix"      default="audio"/>      

  <arg name="nodelet_manager"   default="$(arg base_name)_logger_manager" if="$(arg start_manager)"/>
  <arg name="nodelet_manager"   default="$(arg base_name)" unless="$(arg start_manager)"/>     
  
  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)"  
        output="screen" args="manager" if="$(arg start_manager)">
    <param name="num_worker_threads" value="4"/>
  </node>

  <!-- Tracking logging node -->
  <node pkg="nodelet" type="nodelet" name="$(arg base_name)_logger"
        args="load gear_data_handler/AudioLogger $(arg nodelet_manager)" output="screen">
    <remap from="/log_audio"          to="/audio/audio" />
    <remap from="/image_count"        to="/audio_count" />
    <param name="sensor_id"           value="$(arg base_name)"/>
    <param name="data_dir"            value="$(arg data_dir)"/>
    <param name="image_extn"          value="$(arg image_extn)"/>
    <param name="image_type"          value="$(arg image_type)"/>
    <param name="image_prefix"        value="$(arg image_prefix)"/>
  </node>

</launch>